### PR TITLE
volgen dump implementation using recursive func and slices.

### PR DESCRIPTION
**Pull request by anekkunt**
Signed-off-by: Anand anekkunt@redhat.com

### CODE REVIEW

File: volgen/volgen.go [hunk 1]
```diff
@@ -1,9 +1,27 @@
...
 )
 
 func (x Xlator) DumpGraph(w io.Writer) {
-	_, _ = w.Write([]byte("Dump function not implemented yet"))
+
+	for _, xl := range x.Children {
+		xl.DumpGraph(w)
+	}
+
+	str := fmt.Sprintf("volume %s\n", x.Name)
```
Comment (kshlm) [inline volgen/volgen.go:14]: You could directly write to the io.Writer using Fprint\* methods.

File: volgen/volgen.go [hunk 2]
```diff
@@ -1,9 +1,27 @@
...
+	str += fmt.Sprintf("   type %s", x.Type)
+
+	if x.Children != nil {
+		str += fmt.Sprintf("\n   subvolumes")
+
+		for _, ch_xl := range x.Children {
+			str += fmt.Sprintf(" %s", ch_xl.Name)
+		}
+	}
```
Comment (kshlm) [inline volgen/volgen.go:23]: Where are the options?

File: volgen/volgen.go (hunk 1 above)
Comment (anekkunt) [inline volgen/volgen.go:14]: ok .. I will do it next patch

File: volgen/volgen.go (hunk 2 above)
Comment (anekkunt) [inline volgen/volgen.go:23]: I will do it next path set.

**Pull request closed without merging**